pkg/controller/jobframework/reconciler.go

```diff
@@ -1085,8 +1085,9 @@ func (r *JobReconciler) startJob(ctx context.Context, job GenericJob, object cli
 			return err
 		}
 	} else {
-		if err := clientutil.Patch(ctx, r.client, object, true, func() (bool, error) {
-			return true, job.RunWithPodSetsInfo(info)
+		if err := clientutil.Patch(ctx, r.client, object, true, func() (client.Object, error) {
+			err := job.RunWithPodSetsInfo(info)
+			return job.Object(), err
```

Comment:
```suggestion
			return object, job.RunWithPodSetsInfo(info)
```
I’m wondering if we can do it like this.

Reply:
it's quite easy to verify